lib/Service/LoginService.php

```diff
@@ -253,6 +260,24 @@ public function login($profile, $userSession, $request)
                 }
             }
 
+            if ($this->config->getSystemValue('oidc_login_update_avatar', false)
+                && \array_key_exists($attr['photoURL'], $profile)
+                && $profile[$attr['photoURL']]) {
+                try {
+                    $curl = curl_init($profile[$attr['photoURL']]);
+                    curl_setopt($curl, CURLOPT_HEADER, false);
+                    curl_setopt($curl, CURLOPT_RETURNTRANSFER, true);
+                    curl_setopt($curl, CURLOPT_BINARYTRANSFER, true);
+                    curl_setopt($curl, CURLOPT_USERAGENT, 'Mozilla/5.0');
+                    $raw = curl_exec($curl);
+                    curl_close($curl);
+
+                    $avatar = $this->avatarManager->getAvatar($user->getUid());
+                    $avatar->set($raw);
```

Comment:
I dug into the implementation of the [`set`](https://github.com/nextcloud/server/blob/9124c69cf6ca7cf7551112f19a9e5431f9cb6f13/lib/private/Avatar/UserAvatar.php#L100) call and found that it allows to load an image from a path (see [`getAvatarImage`](https://github.com/nextcloud/server/blob/9124c69cf6ca7cf7551112f19a9e5431f9cb6f13/lib/private/Avatar/UserAvatar.php#L141-L145)). So if the profile photo URL returns a string like `/path/to/some/image`, the functions will happily load the image and set it as an avatar. However, as the code does certain checks like restricting the image to `png` and `jpg` as well as requiring a perfectly square image, the file inclusion is rather limited here. Nevertheless, there might be cases (although special) where someone would be able to include an image from another user (and thus leak it) if the location on the filesystem is known and the image is square.

So maybe we should rather load the image ourself to avoid all the other cases the `getAvatarImage` has (like the local file inclusion).

```suggestion
                    $image = new \OC_Image();
                    $image->loadFromData($raw);
                    
                    $avatar = $this->avatarManager->getAvatar($user->getUid());
                    $avatar->set($image);
```

Reply:
Good catch!

Reply:
Actually `use \OC_Image;` is not needed, and it makes php-cs-linter yell.

Reply:
> Actually use \OC_Image; is not needed, and it makes php-cs-linter yell.

Oh, yeah, makes sense ^^